pkgs/development/libraries/wlroots/default.nix

```diff
@@ -0,0 +1,135 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkg-config
+, wayland-scanner
+, libGL
+, wayland
+, wayland-protocols
+, libinput
+, libxkbcommon
+, pixman
+, libcap
+, mesa
+, xorg
+, libpng
+, ffmpeg_4
+, hwdata
+, seatd
+, vulkan-loader
+, glslang
+, nixosTests
+
+, enableXWayland ? true
+, xwayland ? null
+}:
+
+let
+  generic = { version, hash, extraBuildInputs ? [ ], extraNativeBuildInputs ? [ ], extraPatch ? "" }:
```

Comment:
I feel like it'd be better to have the generic builder in a separate file rather than a function in a let binding.

Reply:
Having a separate file (e.g., `generic.nix`) might indeed be more common in Nixpkgs. I'm fine with it either way though. One technical reason against the current approach is that `generic` isn't accessible (could be useful for overriding and packaging other versions in overlays, etc. but it isn't required and one cannot rely on it anyway since we might perform breaking changes in the future - so it might actually be for the best that it isn't accessible). Anyway, we could also change this at any point in the future so unless there are good (technical) arguments for a different approach I'd leave it up to @peterhoeg for now.

The current version LGTM - I'll do a proper review tomorrow (or rather later today... :o :D).